conjure-go-contract/errors/generic_error.go

```diff
@@ -76,16 +77,26 @@ func (e genericError) InstanceID() uuid.UUID {
 	return e.errorInstanceID
 }
 
+func (e genericError) SafeParams() map[string]interface{} {
+	// Add errorInstanceId as safe param
+	idStorer := wparams.NewSafeParamStorer(map[string]interface{}{"errorInstanceId": e.errorInstanceID})
+	return wparams.NewParamStorer(e.params, idStorer).SafeParams()
```

Comment:
I was trying to avoid mutating the underlying map, and `NewParamStorer` copies everything. I could also do that in here with something like
```go
func (e genericError) SafeParams() map[string]interface{} {
	// Copy safe params map (so we don't mutate the underlying one) and add errorInstanceId
	safeParams := make(map[string]interface{}, len(e.params.SafeParams())+1)
	for k, v := range e.params.SafeParams() {
		safeParams[k] = v
	}
	safeParams["errorInstanceId"] = e.errorInstanceID
	return safeParams
}
```
thoughts?

The reason mutating it is undesirable is it will end up in the marshalled payload

Reply:
Got it, sorry, forgot that `SafeParams()` didn't copy things. Yeah, if you inlined the contents of the current code and your proposal above, the actual diff is fairly small so fine with either the original or your slightly more efficient proposal in the comment.